e2e/stack_test.go

```diff
@@ -99,6 +104,58 @@ func TestE2E(t *testing.T) {
 
 	const targetHeight = 5
 
+	// Hang until L1 responsive.
+	require.True(t, stackConfig.L1URL.IsReachable(ctx))
+
+	l1RPCClient, err := rpc.DialContext(ctx, stackConfig.L1URL.String())
+	require.NoError(t, err)
+	l1Client := e2e.NewL1Client(l1RPCClient)
+
+	// instantiate L1 user, tx signer.
+	user := stackConfig.Users[0]
+	signer := types.NewEIP155Signer(stackConfig.L1ChainID)
+
+	// op Portal
+	portal, err := bindings.NewOptimismPortal(stackConfig.DepositContractAddress, l1Client)
+	require.NoError(t, err)
+
+	// send user Deposit Tx
+	nonce, err := l1Client.Client.NonceAt(ctx, user.Address, nil)
+	require.NoError(t, err)
+
+	price, err := l1Client.Client.SuggestGasPrice(context.Background())
+	require.NoError(t, err)
+
+	gasPrice := new(big.Int).Mul(price, big.NewInt(2))
+	if gasPrice.BitLen() > 256 {
+		gasPrice = price // fallback to original price if overflow
+	}
+
+	depositTx, err := portal.DepositTransaction(
+		&bind.TransactOpts{
+			From: user.Address,
+			Signer: func(addr common.Address, tx *types.Transaction) (*types.Transaction, error) {
+				signed, err := types.SignTx(tx, signer, user.PrivateKey)
+				if err != nil {
+					return nil, err
+				}
+				return signed, nil
+			},
+			Nonce:    big.NewInt(int64(nonce)),
+			GasPrice: big.NewInt(price.Int64() * 2),
+			GasLimit: 1e6,
+			Value:    big.NewInt(1e18 / 10), // 0.1 eth
+			Context:  ctx,
+			NoSend:   false,
+		},
+		user.Address,
+		big.NewInt(1e18/20), // 0.5 eth deposit to L2
```

Comment:
Readbility might be better if you extract `big.NewInt(1e18/20)` out to a contant

Reply:
b169ea0, ec36f24

These are mostly dummy values. I extracted `oneEth` and used it directly.